pkg/datapath/linux/ipsecrps/ipsec_rps_linux.go

```diff
@@ -0,0 +1,142 @@
+		xdpEnabler = xdp.NewEnabler(
+			xdp.AccelerationModeNative,
+			// Cause startup to bail if XDP cannot be enabled in native mode.
+			// Native is required for performance reasons.
+			xdp.WithEnforceXDPNative("required for accelerated IPSec with RPS"),
+		)
+		cpumapEnabler = cpumap.NewEnabler(true)
+	}
+
+	return newUserCfgOut{
+		UserCfg:       cfg,
+		XDPEnabler:    xdpEnabler,
+		CPUMapEnabler: cpumapEnabler,
+	}
+}
+
+// Config is the materialized IPSec RPS configuration.
+type Config struct {
+	enabled bool
+}
+
+// newConfigIn represents the dependencies required by the IPSec RPS configuration.
+type newConfigIn struct {
+	cell.In
+	// Dcfg is the (legacy) materialized daemon configuration. This is used to
+	// read the encryption mode and validate IPSec is enabled.
+	Dcfg *option.DaemonConfig
+	// XDPCfg is the materialized XDP configuration. This is used to ensure that
+	// the materialized XDP config is native. This check **should** be provided by
+	// the validator which is added to the enabler returned from newUserCfg, but
+	// it's an easy check to perform to catch a potential bug. Regardless, this
+	// is here to inform the dependency graph that the XDP config needs to be
+	// determined before the IPSec RPS config can be created.
+	XDPCfg xdp.Config
+	// TunnelCfg is the materialized tunneling configuration. This is used to ensure
+	// that tunneling is disabled, as IPSec RPS is not currently compatible
+	// with any form of tunneling enabled within the datapath.
+	TunnelCfg tunnel.Config
+	// CPUMapCfg is the materialized CPUMap configuration. This is used to ensure
+	// that the CPUMap has been enabled, as the IPSec RPS feature cannot function
+	// without it.
+	CPUMapCfg cpumap.Config
+	// Cfg is the user provided configuration options. This is used to determine
+	// if the user has requested IPSec RPS to be enabled.
+	Cfg userCfg
+}
+
+func newConfig(in newConfigIn) (Config, error) {
+	if !in.Cfg.Enabled {
+		return Config{enabled: false}, nil
+	}
+
+	// Assertions
+	// Is IPSec enabled?
+	if !in.Dcfg.EncryptionEnabled() || !in.Dcfg.EnableIPSec {
+		return Config{}, fmt.Errorf("ipsec rps can only be enabled if IPSec is enabled")
+	}
+	// Is the XDP mode native?
+	if in.XDPCfg.Disabled() || in.XDPCfg.AccelerationMode() != xdp.AccelerationModeNative {
+		return Config{}, errors.New("ipsec rps can only be enabled if XDP is enabled in native mode")
+	}
+	// Is the CPUMap enabled?
+	if !in.CPUMapCfg.Enabled() {
+		return Config{}, errors.New("ipsec rps can only be enabled if the cpu map is enabled")
+	}
+	// Is tunneling disabled?
+	if in.TunnelCfg.Protocol() != tunnel.Disabled {
+		return Config{}, errors.New("ipsec rps can only be enabled if tunneling is disabled")
+	}
+
+	return Config{enabled: true}, nil
+}
+
+func (cfg Config) Enabled() bool { return cfg.enabled }
+
+func (cfg Config) datapathConfigProvider() dpcfgdef.NodeOut {
+	defines := make(dpcfgdef.Map)
+
+	if cfg.Enabled() {
+		defines["ENABLE_IPSEC_RPS"] = "1"
```

Comment:
Feel free to make this a `DECLARE_CONFIG(bool, enable_ipsec_rps, "...")` at the top of bpf_xdp.c and specify it as a constant during XDP loading. I'll have more to share soon, but this will automatically include it in datapath config scaffolding codegen that we'll migrate all configurables to over time.